packages/skill-lesson/hooks/use-next-lesson.ts

```diff
@@ -0,0 +1,32 @@
+import {SanityDocument} from '@sanity/client'
+import {useRouter} from 'next/router'
+
+import {type LessonResource} from '../schemas/lesson-resource'
+
+import {useQuery} from '@tanstack/react-query'
+
+export const useNextLesson = (
+  lesson: LessonResource,
+  module: SanityDocument,
+  section?: SanityDocument,
+) => {
+  const router = useRouter()
+  const {data: nextExercise} = useQuery(
+    ['next-lesson', lesson.slug],
+    async () => {
+      return await fetch('/api/skill/lessons/next', {
+        method: 'POST',
+        body: JSON.stringify({
+          type: lesson._type,
+          slug: router.query.exercise as string,
+          module: module.slug.current,
+          section: section?.slug,
+        }),
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      }).then((r) => r.json())
+    },
+  )
```

Comment:
the api routes are local to total typescript for now but can migrate to `skill-api`

also having those routes independent might give us the ability to perform different logic per product if we wanted/needed to